salt/_modules/caasp_kubernetes_secrets.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import absolute_import
+
+from hashlib import sha1
+from tempfile import mkstemp
+import os
+
+
+def __virtual__():
+    return "caasp_kubernetes_secrets"
+
+
+def kubectl(command, **kwargs):
+    return __salt__['caasp_kubernetes_kubectl.kubectl'](command, **kwargs)
+
+
+def temp_file_with_contents(contents):
+    fd, fp = mkstemp()
+    temp_file = open(fp, 'w')
+    temp_file.write(contents)
+    os.close(fd)
+    return fp
+
+
+def name_by_content(prefix, content):
+    return prefix + sha1(content).hexdigest()
+
+
+def present_with_contents(secret_key, secret_contents, secret_name_prefix=None, secret_name=None, namespace="kube-system"):
+    secret_name = name_by_content(secret_contents) if secret_name is None else secret_name
+    if secret_name_prefix is not None:
+        secret_name = secret_name_prefix + secret_name
+    secret_file_path = temp_file_with_contents(secret_contents)
+    kubectl('create secret generic %s --from-file=%s=%s' % (secret_name, secret_key, secret_file_path), namespace=namespace)
+    os.remove(secret_file_path)
```

Comment:
Instead of using this pattern for creating a temporary file, I think it would be better to have a `kubectl_apply` function that could accept maybe two different parameters, `file=` and `string=`. Then, when using a `string`, we could use the better Python idiom:

```Python
with tempfile.TemporaryFile() as fp:
    fp.write("the manifest contents")
    # run the kubectl apply now
```

Reply:
Oh, do you suggest to use `kubectl apply` to create the secrets too? We can do it that way, yes. In that case I could reuse the `caasp_kubernetes_kubectl.kubectl` logic that doesn't use a temporary file but `stdin` instead.